### PR TITLE
Add memberlist cluster label support to jsonnet

### PR DESCRIPTION
**Pull request by pracucci**
#### What this PR does
This PR adds memberlist cluster label support to jsonnet, assuming https://github.com/grafana/dskit/pull/187 will be accepted.

To check the config changes during the migration process:
```
diff operations/mimir-tests/test-memberlist-cluster-label-migration-step-0-before-generated.yaml operations/mimir-tests/test-memberlist-cluster-label-migration-step-1-generated.yaml
diff operations/mimir-tests/test-memberlist-cluster-label-migration-step-1-generated.yaml operations/mimir-tests/test-memberlist-cluster-label-migration-step-2-generated.yaml
diff operations/mimir-tests/test-memberlist-cluster-label-migration-step-2-generated.yaml operations/mimir-tests/test-memberlist-cluster-label-migration-step-3-generated.yaml
```

#### Which issue(s) this PR fixes or relates to
N/A

#### Checklist

- [ ] Tests updated
- [ ] Documentation added
- [x] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

File: operations/mimir-tests/test-memberlist-cluster-label-migration-step-2.jsonnet [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+    ruler_enabled: true,
+    ruler_client_type: 'gcs',
+    ruler_storage_bucket_name: 'rules-bucket',
+
+    alertmanager_enabled: true,
+    alertmanager_client_type: 'gcs',
+    alertmanager_gcs_bucket_name: 'alerts-bucket',
+
+    // Step 1: Set the cluster label.
```
Comment (replay) [inline operations/mimir-tests/test-memberlist-cluster-label-migration-step-2.jsonnet:21]:
```suggestion
    // Step 2: Set the cluster label.
```

File: operations/mimir-tests/test-memberlist-cluster-label-migration-step-3.jsonnet [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+    ruler_enabled: true,
+    ruler_client_type: 'gcs',
+    ruler_storage_bucket_name: 'rules-bucket',
+
+    alertmanager_enabled: true,
+    alertmanager_client_type: 'gcs',
+    alertmanager_gcs_bucket_name: 'alerts-bucket',
+
+    // Step 1: Re-enable the cluster label verification.
```
Comment (replay) [inline operations/mimir-tests/test-memberlist-cluster-label-migration-step-3.jsonnet:21]:
```suggestion
    // Step 3: Re-enable the cluster label verification.
```

File: operations/mimir/memberlist.libsonnet [hunk 3]
```diff
@@ -2,7 +2,15 @@
   local memberlistConfig = {
     'memberlist.bind-port': gossipRingPort,
     'memberlist.join': 'gossip-ring.%s.svc.cluster.local:%d' % [$._config.namespace, gossipRingPort],
-  },
+  } + (
+    if $._config.memberlist_cluster_label == '' then {} else {
+      'memberlist.cluster-label': $._config.memberlist_cluster_label,
+    }
+  ) + (
+    if $._config.memberlist_cluster_label_verification_enabled then {} else {
```
[the remark this answers is not in the document]
Comment (pracucci) [reply operations/mimir/memberlist.libsonnet]: Done.

**Comment by pracucci** [comment]
@replay Thanks for your review. Could you check it again, please?

**Comment by replay** [comment]
I think the configuration in the `4` steps in `operations/mimir-tests/test-memberlist-cluster-label-migration-step-*` make sense and they look correct to me.

Just to make sure I understand correctly how you're planning to use them, these steps are only used to test the migration scenario in some test environment and verify that the migration works, correct? Or are you planning to also include the configs in these steps files in any prod jsonnet?

**Comment by pracucci** [comment]
> Just to make sure I understand correctly how you're planning to use them, these steps are only used to test the migration scenario in some test environment and verify that the migration works, correct? Or are you planning to also include the configs in these steps files in any prod jsonnet?

The tests I've written are just to test the config changes during different steps of the migration. My plan to run the migration is to open a PR for each step.

**Review by replay — APPROVED** [review]
Looks great, thanks!

**Pull request merged**